src/events/base.py

```diff
@@ -68,6 +70,13 @@ def get_app_status(  # noqa: C901
         if self.charm.app.planned_units() > 1 and not self.context.zookeeper:
             return Status.MISSING_ZOOKEEPER.value
 
+        if self.charm.config.get("expose-external", "false") not in VALID_EXPOSE_EXTERNAL_VALUES:
+            return Status.INVALID_EXPOSE_EXTERNAL.value
+
+        service_util = ServiceUtil(self.charm.model)
```

Comment:
When computing the status (which happens immediately after `config-changed` and all other major hooks), we check the connectivity of the service endpoints and put the charm in Maintenance state. Later on when the deferred config-changed hook or any other hook uses `@compute_status` decorator, the connectivity is checked again and if found connectible, charm is put into active state again.